src/components/Rubric.js

```diff
@@ -1,177 +1,171 @@
 
-const CompetencyButton = props => {
+const CompetencyButton = withRouter(props => {
   return (
     <li className={props.isActive}
       onClick={() => {
         props.setActiveComp(props.index)
       }}>
-      <a href="#dashboard-2" role="tab" data-toggle="tab" aria-expanded="true">
+      <a href="" onClick={(e) => e.preventDefault()} role="tab" data-toggle="tab" aria-expanded="true">
         <i className="material-icons">{props.icon}</i> {props.name}
-      </a>
-    </li>
-  )
-}
-
-const Criteria = props => {
-  return (
-    <tr>
-      <td>
-        <div className="checkbox">
-          <label>
-            <input type="checkbox" name="optionsCheckboxes" defaultChecked={false} />
-          </label>
-        </div>
-      </td>
-      <td>{props.text}</td>
-    </tr>
-  )
-}
-
-class Rubric extends Component {
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      activeCompetencyIndex: 0
+        </a>
+      </li>
+    )
+  })
+
+  class Rubric extends Component {
+
+    constructor(props) {
+      super(props);
+      this.state = {
+        activeCompetencyIndex: 0
+      }
+      this.icons = [ "dashboard", "explore", "code", "backup", "lock", "bug_report", "line_style", "perm_identity", "star_rate" ]
+      this.renderCompetencies = this.renderCompetencies.bind(this);
+      this.renderCompetencyButtons = this.renderCompetencyButtons.bind(this);
+      this.renderCriteriaForLevel = this.renderCriteriaForLevel.bind(this);
+      this.renderLevels = this.renderLevels.bind(this);
+      this.getIsFetching = this.getIsFetching.bind(this);
     }
-    this.icons = [ "dashboard", "explore", "code", "backup", "lock", "bug_report", "line_style", "perm_identity", "star_rate" ]
-    this.renderCompetencies = this.renderCompetencies.bind(this);
-    this.renderCompetencyButtons = this.renderCompetencyButtons.bind(this);
-    this.renderCriteriaForLevel = this.renderCriteriaForLevel.bind(this);
-    this.renderLevels = this.renderLevels.bind(this);
-    this.getIsFetching = this.getIsFetching.bind(this);
-  }
 
-  componentWillMount() {
-    const id = this.props.match.params.id;
-    this.props.getAssessment(localStorage.getItem("userId"), id);
-  }
-
-  setActiveComp(index) {
-    this.setState({
-      activeCompetencyIndex: index
-    })
-  }
+    componentWillMount() {
+      const id = this.props.match.params.id;
+      this.props.getAssessment(localStorage.getItem("userId"), id);
+    }
 
-  renderCompetencyButtons() {
-    if (this.props.assessment.rubricJSON) {
-      return this.props.assessment.rubricJSON.Competencies.map((comp, index) => {
-        var active = "";
-        if (index === this.state.activeCompetencyIndex) {
-          active = "active";
-        }
-        return (
-          this.getIsFetching() ? "" :
-            <CompetencyButton
-              name={comp.name}
+    renderCriteriaForLevel(level, compIndex) {
+      if (this.props.assessment.assessmentObject.rubricJSON.Competencies) {
```

Comment:
deconstructing this would make the code so much more readable. 

`const { Competencies } = this.props.assessment.assessmentObject.rubricJSON;`

Then access only the variable. Saves you having to write `this.props.assessment.assessmentObject.rubricJSON` twice!